src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -62,6 +64,247 @@ static ShardPlacement * TupleToShardPlacement(TupleDesc tupleDesc,
+	char *schemaName = get_namespace_name(schemaId);
+
+	List *workerNodeList = WorkerNodeList();
+	ListCell *workerNodeCell = NULL;
+
+	int shardIndex = 0;
+	uint64 totalRelationSize = 0;
+
+	MultiConnection *connection = NULL;
+
+	/* Check whether the size of table can be obtained with citus size functions. */
+	ErrorIfNotSuitableToGetSize(distributedRelationId);
+
+	distributedTableCacheEntry = DistributedTableCacheEntry(distributedRelationId);
+
+	foreach(workerNodeCell, workerNodeList)
+	{
+		WorkerNode *workerNode = (WorkerNode *) lfirst(workerNodeCell);
+		char *workerNodeName = workerNode->workerName;
+		uint32 workerNodePort = workerNode->workerPort;
+		bool shardSeenOnNode = false;
+		PGresult *queryResult = NULL;
+		int executeCommand = 0;
+		char *tableSizeString
+
+		StringInfo tableSizeQuery = makeStringInfo();
+		appendStringInfoString(tableSizeQuery, "Select ");
+
+		for (shardIndex = 0; shardIndex <
+			 distributedTableCacheEntry->shardIntervalArrayLength; shardIndex++)
+		{
+			ShardPlacement *placement =
+				distributedTableCacheEntry->arrayOfPlacementArrays[shardIndex];
+			char *shardNodeName = placement->nodeName;
+			uint32 shardNodePort = placement->nodePort;
+
+			/*Reference table exist on each node*/
+			if ((strcmp(shardNodeName, workerNodeName) == 0 && shardNodePort ==
+				 workerNodePort) ||
+				distributedTableCacheEntry->partitionMethod == DISTRIBUTE_BY_NONE)
+			{
+				uint64 shardId = placement->shardId;
+				char *shardName = get_rel_name(distributedRelationId);
+				char *shardQualifiedName = NULL;
+				char *quotedShardName = NULL;
+				quote_literal_cstr(shardQualifiedName);
+				AppendShardIdToName(&shardName, shardId);
+
+				shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
+				quotedShardName = quote_literal_cstr(shardQualifiedName);
+
+				if (!shardSeenOnNode)
+				{
+					shardSeenOnNode = !shardSeenOnNode;
+				}
+				else
+				{
+					appendStringInfo(tableSizeQuery, " + ");
+				}
+
+				if (sizeFunctionType == CITUS_TOTAL_RELATION_SIZE)
+				{
+					appendStringInfo(tableSizeQuery, "pg_total_relation_size(%s)",
+									 quotedShardName);
+				}
+				else if (sizeFunctionType == CITUS_TABLE_SIZE)
+				{
+					appendStringInfo(tableSizeQuery, "pg_table_size(%s)",
+									 quotedShardName);
+				}
+				else if (sizeFunctionType == CITUS_RELATION_SIZE)
+				{
+					appendStringInfo(tableSizeQuery, "pg_relation_size(%s)",
+									 quotedShardName);
+				}
+
+				/*For reference table, there is only one shard placement for each node */
+				if (distributedTableCacheEntry->partitionMethod == DISTRIBUTE_BY_NONE)
+				{
+					appendStringInfo(tableSizeQuery, ";");
+					break;
+				}
+			}
+		}
+
+		/* There is no shard placement for this node */
+		if (!shardSeenOnNode)
+		{
+			continue;
+		}
+
+		appendStringInfo(tableSizeQuery, ";");
+
+
+		connection = GetNodeConnection(0, workerNodeName, workerNodePort); /* TODO : I'm not sure about the flag ! */
```

Comment:
I think the flag is correct. Just assign it to a variable. This way a person who reads the code might get confused about what "0" means.